test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -93,6 +100,55 @@ static public void testCreatePoint2D() throws Throwable {
 		assertEquals(getY.invoke(point2D), yNew);
 	}
 
+	/*
+	 * Test array of Point2D
+	 */
+	@Test(priority=2)
+	static public void testCreateArrayPoint2D() throws Throwable {
+		MethodHandle withX = generateWither(point2DClass, "x", int.class);
+		MethodHandle withY = generateWither(point2DClass, "y", int.class);
+
+		int x1 = 0xFFEEFFEE;
+		int y1 = 0xAABBAABB;
+		int x1New = 0x11223344;
+		int y1New = 0x99887766;
+		
+		Object point2D_1 = makePoint2D.invoke(x1, y1);
+		
+		assertEquals(getX.invoke(point2D_1), x1);
+		assertEquals(getY.invoke(point2D_1), y1);
+		
+		point2D1 = withX.invoke(point2D_1, x1New);
+		point2D1 = withY.invoke(point2D_1, yNew1);
+		
+		assertEquals(getX.invoke(point2D_1), x1New);
+		assertEquals(getY.invoke(point2D_1), y1New);
+
+
+		int x2 = 0x00000011;
+		int y2 = 0xABCDEF00;
+		int x2New = 0x11111111;
+		int y2New = 0xFFFFFFFF;
+		
+		Object point2D_2 = makePoint2D.invoke(x2, y2);
+		
+		assertEquals(getX.invoke(point2D_2), x2);
+		assertEquals(getY.invoke(point2D_2), y2);
+		
+		point2D1 = withX.invoke(point2D_2, x2New);
+		point2D1 = withY.invoke(point2D_2, y2New);
+		
+		assertEquals(getX.invoke(point2D_2), x2New);
+		assertEquals(getY.invoke(point2D_2), y2New);
+
+		Class arrayClass = Array.newInstance(point2DClass, 3).getClass;
+		Array.set(arrayClass, 1, point2D_1);
+		Array.set(arrayClass, 2, point2D_2);
+
+		assertEquals(Array.get(arrayClass, 1), point2D_1);
```

Comment:
this will not work with flattened arrays as it is doing an identity comparison. Need to check the elements individually